src/execution.jl

```diff
@@ -334,8 +312,97 @@ end
 
 ## host-side API
 
-const agecache = Dict{UInt, UInt}()
+using Core.Compiler: retrieve_code_info, CodeInfo, MethodInstance, SSAValue, SlotNumber
+using Base: _methods_by_ftype
+
+# HACK: mechanism to generate calls that are not executed, but ensure method invalidation
+const opaque_false = Ref(false)
+function fake_call(f)
+    opaque_false[] || return
+    f(Ref{Any}()[]...)
+end
+
+# actual compilation
+function cufunction_slow(f, tt, spec; name=nothing, kwargs...)
+    start = time_ns()
+
+    # generate a fake call to ensure we get recompiled upon method invalidation
+    fake_call(f)
```

Comment:
Why does this function need to be recompiled?

Reply:
Ah, right. I expected the entire call chain to get recompiled, which is wrong of course. So this should be part of the generator output.